File: 2192 All Ancestors of a Node in a Directed Acyclic Graph/1550 Three Consecutive Odds/1550.cpp
class Solution {
public:
    bool threeConsecutiveOdds(vector<int>& arr) {
        for (int i = 2; i < arr.size(); i += 3) {
            if (arr[i] % 2 == 1) {
                if (arr[i - 1] % 2 == 1) {
                    if (arr[i - 2] % 2 == 1) return true;
                    if (i + 1 < arr.size() && arr[i + 1] % 2 == 1) return true;
                    else { i++; continue; }
                }
                if (i + 2 < arr.size() && arr[i + 1] % 2 == 1) {
                    if (arr[i + 2] % 2 == 1) return true;
                    else {i += 2; continue;}
                }
                else { i++; continue;}
            }
        }
        return false;
    }

};

File: 2192 All Ancestors of a Node in a Directed Acyclic Graph/1550 Three Consecutive Odds/1550.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
var threeConsecutiveOdds = function(arr) {
    for (let i=2; i<arr.length; i+=3) {
        if (arr[i] % 2) {
            if (arr[i-1] % 2) {
                if (arr[i-2] % 2) return true;
                if (i+1 < arr.length && arr[i+1] % 2) return true;
                else { i++; continue; }
            }
            if (i+2 < arr.length && arr[i+1] % 2) {
                if (arr[i+2] % 2) return true;
                else {i+=2; continue;}
            }
            else { i++; continue;}
        }
    }
    return false;
};
